packages/arb-token-bridge-ui/src/util/withdrawals/fetchTokenWithdrawalsFromEventLogsSequentially.ts

```diff
@@ -0,0 +1,151 @@
+import { constants } from 'ethers'
+import { Provider, BlockTag } from '@ethersproject/providers'
+import { Erc20Bridger, getArbitrumNetwork } from '@arbitrum/sdk'
+
+import {
+  fetchTokenWithdrawalsFromEventLogs,
+  FetchTokenWithdrawalsFromEventLogsParams
+} from './fetchTokenWithdrawalsFromEventLogs'
+import { getNonce } from '../AddressUtils'
+import { fetchL2Gateways } from '../fetchL2Gateways'
+
+function wait(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function getGateways(provider: Provider): Promise<{
+  standardGateway: string
+  wethGateway: string
+  customGateway: string
+  otherGateways: string[]
+}> {
+  const network = await getArbitrumNetwork(provider)
+
+  const standardGateway = network.tokenBridge?.childErc20Gateway
+  const customGateway = network.tokenBridge?.childCustomGateway
+  const wethGateway = network.tokenBridge?.childWethGateway
+  const otherGateways = await fetchL2Gateways(provider)
+
+  return {
+    standardGateway: standardGateway ?? constants.AddressZero,
+    wethGateway: wethGateway ?? constants.AddressZero,
+    customGateway: customGateway ?? constants.AddressZero,
+    otherGateways
+  }
+}
+
+type UnwrapPromise<T> = T extends Promise<infer U> ? U : T
```

Comment:
Wouldn't built-in [Awaited](https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype) do the same thing?